### ISSUE TITLE
Add PanelApp as a source

### ISSUE THREAD

**Issue by mvelinder** [iobio/clin.iobio#195]
https://panelapp.genomicsengland.co.uk/

**Comment by AlistairNWard** [comment]
Should this appear under clin or genepanel. I see you added it to both (and I closed the one on genepanel since I added this in #24).

Maybe we just need to go through issues on the different tools and prioritize.

**Comment by adityaekawade** [comment]
I was experimenting in genepanel a couple of months back how PanelApp can be integrated. There are a limited number of panels reviewed in the PanelApp (currently 324), so instead of creating a workflow similar to GTR, phenolyzer; we detect if a selected term from these tools is reviewed in the PanelApp. 
Consider the following workflow: 
<img width="831" alt="Screen Shot 2020-04-27 at 11 23 27 AM" src="https://user-images.githubusercontent.com/16284713/80415342-46529080-8887-11ea-9ea2-41e89a3fa227.png">


Test: 

1. Enter the url https://dev.panel.iobio.io/ and go to "Dashboard". 
2. Select the term "Amyloidosis" from the type ahead 
3. Select the general term "Amyloidosis" for each tool and generate gene list.
4. The top 7 genes have a green "reviewed" icon. These are the high evidence genes from PanelApp for  "Amyloidosis". (Ref: https://panelapp.genomicsengland.co.uk/panels/502/)
 
<img width="919" alt="Screen Shot 2020-04-27 at 1 12 53 PM" src="https://user-images.githubusercontent.com/16284713/80416365-dcd38180-8888-11ea-9eac-db8a9d8fd01a.png">

We can expand on how we want to display this information but here is a basic example of how PanelApp can be integrated in the existing workflow. Let me know what you think.

**Comment by AlistairNWard** [comment]
Just so I understand, the rationale here is that if we used the same setup as e.g. GTR, most of the time, there would be nothing to select for PanelApp?

Just my stream of consciousness thinking here. Let's say you do the example above, and instead of picking the general term, I picked other terms (e.g. AFib Amyloidosis in GTR etc.) This results in no hits for PanelApp, and the user wouldn't know that the PanelApp did have info for some of the other terms. We could:
* Stick with the same workflow as in GTR, Phenolyzer and accept there is often nothing to select
* Use the proposal above and only get info from PanelApp if you picked a term that was present
* Use the proposal above, but when picking from the GTR list, indicate which terms are present in PanelApp in this list. You might then choose a general term knowing that there is also a PanelApp hit

Also, we should consider scalability. How quickly is PanelApp growing? If a year from now there are way more terms in PanelApp, would we want the same workflow as with GTR?

These are all just things that come to mind - I'm not suggesting anything!

**Comment by adityaekawade** [comment]
> Just so I understand, the rationale here is that if we used the same setup as e.g. GTR, most of the time, there would be nothing to select for PanelApp?

Yes exactly! Also if there is a hit, the options for selection would be few. The proposed solution would let the user make fewer selections but fetch data from an additional source. 


The NLP used for processing the note can detect the general term. So even if the user selects "AFib Amyloidosis" we know that there is a general term (Amyloidosis) present and there is a hit in the PanelApp. So at the time of the review, we can show the terms that are present in PanelApp from the selected terms. 

> Also, we should consider scalability. How quickly is PanelApp growing? If a year from now there are way more terms in PanelApp, would we want the same workflow as with GTR?

I had created this doc (https://docs.google.com/document/d/1136Ldl9AyPcNHOkZ2qenO3ezKuKfb9A-r8GqWp5MXqY/edit#heading=h.f9bwhm6l23v1) in September 2018. There were 215 Panels at that time and 324 now. So about 110 new panels are added in one and half years. 

> If you search in PanelApp for Amyloidosis it has 11 associated genes, but the proposal only shows 7 of them because it's relying on GTR etc to also find the gene. This doesn't seem like a good thing!

This was an example and I was ranking the genes that were already present from other sources and I considered only high evidence (green) genes. 
However, The prioritization of the genes will be done based on PanelApp, like you said this is a highly curated set of genes. Here `APOC2` along with the other moderate evidence gene is missing, which will be added to the final list.

**Comment by AlistairNWard** [comment]
>> Just so I understand, the rationale here is that if we used the same setup as e.g. GTR, most of the time, there would be nothing to select for PanelApp?

> Yes exactly! Also if there is a hit, the options for selection would be few. The proposed solution would let the user make fewer selections but fetch data from an additional source.

>The NLP used for processing the note can detect the general term. So even if the user selects "AFib Amyloidosis" we know that there is a general term (Amyloidosis) present and there is a hit in the PanelApp. So at the time of the review, we can show the terms that are present in PanelApp from the selected terms.

Ok, cool. That's what I thought. If we show results for a general term, even if a more specific term wass selected, that would remove my concern here.

>> Also, we should consider scalability. How quickly is PanelApp growing? If a year from now there are way more terms in PanelApp, would we want the same workflow as with GTR?

> I had created this doc (https://docs.google.com/document/d/1136Ldl9AyPcNHOkZ2qenO3ezKuKfb9A-r8GqWp5MXqY/edit#heading=h.f9bwhm6l23v1) in September 2018. There were 215 Panels at that time and 324 now. So about 110 new panels are added in one and half years.

That's what I thought, so again, not a concern.

>> If you search in PanelApp for Amyloidosis it has 11 associated genes, but the proposal only shows 7 of them because it's relying on GTR etc to also find the gene. This doesn't seem like a good thing!

> This was an example and I was ranking the genes that were already present from other sources and I considered only high evidence (green) genes.
> However, The prioritization of the genes will be done based on PanelApp, like you said this is a highly curated set of genes. Here APOC2 along with the other moderate evidence gene is missing, which will be added to the final list.

So, with these comments, I'm pretty happy with this setup

**Comment by AlistairNWard** [comment]
I think this would be worth doing @adityaekawade as proposed.
